gravitee-plugin-annotation-processors/src/test/java/io/gravitee/plugin/annotation/processor/ConfigurationEvaluatorGeneratedTest.java

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.plugin.annotation.processor;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import io.gravitee.gateway.reactive.api.ExecutionFailure;
+import io.gravitee.gateway.reactive.api.context.ExecutionContext;
+import io.gravitee.plugin.annotation.processor.result.SecurityProtocol;
+import io.gravitee.plugin.annotation.processor.result.TestConfiguration;
+import io.gravitee.plugin.annotation.processor.result.TestConfigurationEvaluator;
+import io.reactivex.rxjava3.core.Completable;
+import io.reactivex.rxjava3.core.Maybe;
+import io.reactivex.rxjava3.observers.TestObserver;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * @author Remi Baptiste (remi.baptiste at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ConfigurationEvaluatorGeneratedTest {
+
+    TestConfigurationEvaluator evaluator;
+
+    @Mock
+    ExecutionContext context;
+
+    @Mock
+    ExecutionContext.TemplateEngine templateEngine;
+
+    @Before
+    public void before() {
+        TestConfiguration configuration = new TestConfiguration();
+        configuration.getConsumer().setEnabled(true);
+        configuration.getConsumer().setAutoOffsetReset("none");
+        evaluator = new TestConfigurationEvaluator(configuration);
+
+        when(context.getTemplateEngine()).thenReturn(templateEngine);
+
+        when(context.interruptWith(any()))
+            .thenAnswer(invocation ->
+                Completable.defer(() ->
+                    Completable.error(new IllegalStateException(((ExecutionFailure) invocation.getArgument(0)).message()))
+                )
+            );
+    }
+
+    @Test
+    public void should_interrupt_with_error_on_validation() {
+        when(templateEngine.eval("none", String.class)).thenReturn(Maybe.just("result"));
+
+        TestObserver<TestConfiguration> testObserver = evaluator.eval(context).test();
+
+        testObserver.assertError(throwable -> {
+            assertThat(throwable).isInstanceOf(IllegalStateException.class);
+            assertThat(throwable.getMessage()).isEqualTo("Invalid configuration");
+            return true;
+        });
+    }
+
+    @Test
+    public void should_return_evaluated_configuration() {
+        when(templateEngine.eval("none", String.class)).thenReturn(Maybe.just("latest"));
+        when(context.getAttribute("gravitee.attributes.endpoint.test.protocol")).thenReturn(SecurityProtocol.SSL);
```

Comment:
Looking at the test I'm wondering if we wouldn't have forgotten a particular use case -> enumeration.
Given the fact that the assign attributes policy can only be `String`, it could be interesting to support enum translation.
For example, `ctx.setAttribute("gravitee.attributes.endpoint.test.protocol", "SSL")` will not work and I'm wondering if we should provide automatic support. What are you thoughts?

Reply:
I have added support for Enum so now this code will work. I change the test to reflect it also.